app/frontend/wallet/shelley/shelley-transaction.ts

```diff
@@ -227,15 +154,66 @@ function ShelleySignedTransactionStructured(txAux, witnesses, meta) {
   }
 }
 
+function ShelleyPoolRegistrationCertificate(certificate: _Certificate) {
+  const {type, poolRegistrationParams: poolParams} = certificate
+  const txPoolRegistrationCertificate = certificate.poolRegistrationParams
+    ? [
+      type,
+      Buffer.from(poolParams.poolKeyHashHex, 'hex'),
+      Buffer.from(poolParams.vrfKeyHashHex, 'hex'),
+      parseInt(poolParams.pledgeStr, 10),
+      parseInt(poolParams.costStr, 10),
+      new Tagged(
+        30,
+        [
+          parseInt(poolParams.margin.numeratorStr, 10),
+          parseInt(poolParams.margin.denominatorStr, 10),
+        ],
+        null
+      ),
+      Buffer.from(poolParams.rewardAccountHex, 'hex'),
+      poolParams.poolOwners.map((ownerObj) => {
+        if (ownerObj.stakingKeyHashHex) {
+          return Buffer.from(ownerObj.stakingKeyHashHex, 'hex')
+        }
+        // else is a path owner and has pubKeyHex
+        return Buffer.from(ownerObj.pubKeyHex, 'hex')
+      }),
+      poolParams.relays.map((relay) => {
+        switch (relay.type) {
+          case 0:
```

Comment:
do we really want hard-coded numbers?

Reply:
As mentioned in the PR description, I wasn't touching the pool registration-related functionality. I just move this into a function as it was. I ll refactor the pool registration.